contracts/facets/EmbalmerFacet.sol

```diff
@@ -34,267 +34,237 @@ contract EmbalmerFacet {
+
+        // Confirm the sarcophagus exists
+        if (sarcophagus.resurrectionTime == 0) {
+            revert LibErrors.SarcophagusDoesNotExist(sarcoId);
+        }
+
+        // Confirm the sarcophagus has not been compromised
+        if (sarcophagus.isCompromised) {
+            revert LibErrors.SarcophagusCompromised(sarcoId);
+        }
+
+        // Confirm the sarcophagus is not buried
+        if (sarcophagus.resurrectionTime == 2**256 - 1) {
+            revert LibErrors.SarcophagusInactive(sarcoId);
+        }
+
+        // Confirm tx sender is embalmer
+        if (sarcophagus.embalmerAddress != msg.sender) {
+            revert LibErrors.SenderNotEmbalmer(msg.sender, sarcophagus.embalmerAddress);
         }
 
-        // Confirm current resurrection time is in future (sarcophagus is rewrappable)
-        if (s.sarcophagi[sarcoId].resurrectionTime <= block.timestamp) {
+        // Confirm resurrection time has not yet passed
+        if (sarcophagus.resurrectionTime <= block.timestamp) {
             revert LibErrors.SarcophagusIsUnwrappable();
         }
 
-        // Confirm that the new resurrection time is in the future
+        // Confirm that new resurrection time is in future
         if (resurrectionTime <= block.timestamp) {
             revert LibErrors.NewResurrectionTimeInPast(resurrectionTime);
         }
 
-
-        // Confirm that the new resurrection time doesn't exceed the sarcophagus's maximumRewrapInterval
-        if (resurrectionTime > block.timestamp + s.sarcophagi[sarcoId].maximumRewrapInterval) {
+        // Confirm that new resurrection time doesn't exceed sarcophagus's maximumRewrapInterval
+        if (resurrectionTime > block.timestamp + sarcophagus.maximumRewrapInterval) {
             revert LibErrors.NewResurrectionTimeTooLarge(resurrectionTime);
         }
 
-        // For each archaeologist on the sarcophagus, transfer their digging fee allocations to them
-        address[] memory bondedArchaeologists = s
-            .sarcophagi[sarcoId]
-            .archaeologists;
-
+        // track total digging fees across all archaeologists on the sarcophagus
         uint256 totalDiggingFees = 0;
 
-        for (uint256 i = 0; i < bondedArchaeologists.length; i++) {
-            // Get the archaeolgist's fee data
-            LibTypes.ArchaeologistStorage memory archaeologistData = LibUtils
-                .getArchaeologist(sarcoId, bondedArchaeologists[i]);
-
-            // Transfer the archaeologist's digging fee allocation to the archaeologist's reward pool
-            s.archaeologistRewards[bondedArchaeologists[i]] += archaeologistData.diggingFee;
+        // pay digging fee to each cursed archaeologist on the sarcophagus
+        address[] storage archaeologistAddresses = sarcophagus.cursedArchaeologistAddresses;
+        for (uint256 i = 0; i < archaeologistAddresses.length; i++) {
+            LibTypes.CursedArchaeologist storage cursedArchaeologist = sarcophagus
```

Comment:
this needs to be checking that the archaeologist hasn't been accused, handled in future PR